spirv_msl.cpp

```diff
@@ -201,17 +201,14 @@ bool CompilerMSL::is_var_runtime_size_array(const SPIRVariable &var) const
 uint32_t CompilerMSL::get_resource_array_size(const SPIRType &type, uint32_t id) const
 {
 	uint32_t array_size = to_array_size_literal(type);
-
-	// If we have argument buffers, we need to honor the ABI by using the correct array size
-	// from the layout. Only use shader declared size if we're not using argument buffers.
-	uint32_t desc_set = get_decoration(id, DecorationDescriptorSet);
-	if (!descriptor_set_is_argument_buffer(desc_set) && array_size)
+	if (array_size)
 		return array_size;
 
-	StageSetBinding tuple = { get_entry_point().model, desc_set,
+	StageSetBinding tuple = { get_entry_point().model,
+		                      get_decoration(id, DecorationDescriptorSet),
 		                      get_decoration(id, DecorationBinding) };
 	auto itr = resource_bindings.find(tuple);
-	return itr != end(resource_bindings) ? itr->second.first.count : array_size;
+	return itr != end(resource_bindings) ? itr->second.first.count : 0;
```

Comment:
IIRC, the problem here is mutable descriptor types or aliases where two aliases have different descriptor counts. If I recall correctly, the code will be screwed up if you try to alias resources with different array sizes. I don't understand what this revert is accomplishing.

A runtime array where VkPipelineLayout has a non-variable count must always be allocated with the size of the descriptor set layout. It should always be safe to use that array size in the shader.
For variable count, there is no such problem.

Reply:
> Several hundred CTS tests use a runtime array in SPIR-V to read from a fixed-size array defined in a Vulkan descriptor set.

Yes, and existing code will resize that runtime array in SPIR-V to match the pipeline layout. If anything, it should be more robust, since our [1] unsized array hack is technically UB.

Reply:
If you intended to revert, can you make that a proper revert-commit? That way I don't have to review extra to see if it's a true revert.

Reply:
You are correct, sir. Existing code works.

Somewhere in my testing, I got mixed up with `get_resource_array_size()`, but it was misdirected.

I've force-pushed the reversion of my changes there.

Reply:
The new commit is good, ye. Thanks for verifying.